utils/secrets_test.go

```diff
@@ -45,3 +45,60 @@ func TestSecretAddRotation(t *testing.T) {
 	}
 
 }
+
+func TestV2SameKey(t *testing.T) {
```

Comment:
perhaps personal preference, but I think this test should be broken down into multiple tests since it feel like each explicit case should be tested on its own in isolation. in part this helps to organize things, but I think it also helps compartmentalize and reduce the chance that state from a previous test is effecting the current test